libs/utils.py

```diff
@@ -173,3 +173,23 @@ def _remove_next_suffix(f):
         filepath = filepath.with_suffix("")
 
     return filepath
+
+
+def read_log_file_and_check_line_exists(log_file, expected_lines: list):
```

Comment:
A bit of a nit pick, but these aren't really "expected lines", are they? (And it's not totally clear just from the function definition what a "line" is - is it a number? A string?)

Maybe a variable name like `expected_prefixes` would be clearer than `expected_lines` - to me, a list of "expected prefixes" is more clearly a list of strings that you expect to find at the start of log file lines, which it looks like is what you're intending.

Reply:
Fantastic! I would have the same question in the future :smiley: